packages/financial-templates-lib/src/price-feed/InsuredBridgePriceFeed.ts

```diff
@@ -0,0 +1,172 @@
+        relay.realizedLpFeePct === parsedAncillaryData.realizedLpFeePct.toString() &&
+        relay.slowRelayer.substr(2).toLowerCase() === parsedAncillaryData.slowRelayer &&
+        relay.depositContract.substr(2).toLowerCase() === parsedAncillaryData.depositContract
+    );
+
+    // TODO: Do we need to handle the case where all of these params are matched and matchedRelay.length > 1?
+    if (matchedRelay.length === 0) {
+      this.logger.debug({
+        at: "InsuredBridgePriceFeed",
+        message: "No relay event found for price request time",
+        priceRequestTime: time,
+      });
+      return this.toBNWei(isRelayValid.No);
+    }
+
+    const relay = matchedRelay[0];
+
+    // Validate that relay matches a deposit struct exactly.
+    const matchedDeposits = this.deposits.filter(
+      (deposit: Deposit) =>
+        deposit.depositId === relay.depositId &&
+        deposit.depositHash === relay.depositHash &&
+        deposit.timestamp === relay.depositTimestamp &&
+        deposit.sender === relay.sender &&
+        deposit.recipient === relay.recipient &&
+        deposit.l1Token === relay.l1Token &&
+        deposit.amount === relay.amount &&
+        deposit.slowRelayFeePct === relay.slowRelayFeePct &&
+        deposit.instantRelayFeePct === relay.instantRelayFeePct &&
+        deposit.quoteTimestamp === relay.quoteTimestamp
+    );
+    if (matchedDeposits.length > 1) {
+      this.logger.error({
+        at: "InsuredBridgePriceFeed",
+        message: "TODO: Handle multiple deposits associated with same relay attempt",
+      });
+      throw new Error("TODO: Handle multiple relays for same price request timestamp");
+    } else if (matchedDeposits.length === 0) {
+      this.logger.debug({
+        at: "InsuredBridgePriceFeed",
+        message: "No deposit event matching relay attempt",
+      });
+      return this.toBNWei(isRelayValid.No);
+    }
+
+    // Validate relays proposed realized fee percentage.
+    const expectedRealizedFeePct = this.l1Client.calculateRealizedLpFeesPctForDeposit(/* matchedDeposits[0] */);
+    if (expectedRealizedFeePct !== relay.realizedLpFeePct) {
+      this.logger.error({
+        at: "InsuredBridgePriceFeed",
+        message: "Matched deposit with relay but realized fee % is incorrect",
+      });
+      return this.toBNWei(isRelayValid.No);
+    }
+
+    // TODO: Do we need to check other parameters like slow relayer address and deposit box contract address? These
+    // are the other params included in the ancillary data by the BridgePool contract.
+
+    // Passed all checks, relay is valid!
+    return this.toBNWei(isRelayValid.Yes);
+  }
+
+  public getLastUpdateTime(): number | null {
+    // TODO.
+    return null;
+  }
+
+  public getLookback(): number | null {
+    // TODO.
+    return null;
+  }
+
+  public getCurrentPrice(): BN | null {
+    // TODO. This doesn't seem appropriate for this pricefeed, perhaps it should always return null.
```

Comment:
This seems like it _could_ be implemented almost the same as the historical one except we could just drop the filtering by timestamp since the request should be unique even without the relay timestamp, right?

Reply:
True! I'll implement. So IIUC this method will rarely if at all be used in production because the current time stamp isn't that predictable?

Reply:
Since this is probably re-uses `getHistoricalPrice` I'll save for follow up PR, I really just want to come to consensus on the main `getHistoricalPrice` implementation

Reply:
> So IIUC this method will rarely if at all be used in production because the current time stamp isn't that predictable?

Yeah, if it's _only_ used by bots that use the historical price, then yes. But, for instance, if we wanted to validate the relay before we sent it in the relay bot, we would probably use this method.

Regardless, for this identifier, I don't think the timestamp actually matters when determining if the relay is valid or not, so this method has a reasonable value that we could return and I think it should be identical to what the historical method returns for the same ancillary data unless I'm missing something.